File: SKLibrary/src/main/java/com/silvericekey/skutilslibrary/base/BaseApplication.java
package com.silvericekey.skutilslibrary.base;

import android.app.Application;

import androidx.multidex.MultiDexApplication;

import com.silvericekey.skutilslibrary.utils.activity.ActivityUtil;
import com.silvericekey.skutilslibrary.utils.log.LoggerHelper;

public class BaseApplication extends MultiDexApplication {
    public static Application app;
    @Override
    public void onCreate() {
        super.onCreate();
        app = this;
        registerActivityLifecycleCallbacks(new ActivityUtil());
        //初始化日志框架
        new LoggerHelper.Builder().build().init();
    }
}

File: SKLibrary/src/main/java/com/silvericekey/skutilslibrary/base/BaseApplication.kt
package com.silvericekey.skutilslibrary.base

import androidx.multidex.MultiDexApplication
import com.blankj.utilcode.util.Utils
import com.facebook.drawee.backends.pipeline.Fresco
import java.util.ArrayList

open class BaseApplication : MultiDexApplication() {
    companion object {
        @JvmStatic
        private var app: BaseApplication? = null

        @JvmStatic
        fun getApp(): BaseApplication = app!!
        var presenters: ArrayList<BasePresenter> = arrayListOf()
    }

    override fun onCreate() {
        super.onCreate()
        app = this
        Utils.init(this)
        Fresco.initialize(this)
    }

    open fun statusChange() {

    }
}